src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/MethodIsGenericMapNode.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+using Internal.Text;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class MethodIsGenericMapNode : ModuleSpecificHeaderTableNode
+    {
+        private MetadataReader _metadata;
+
+        public MethodIsGenericMapNode(EcmaModule module) : base(module)
+        {
+            _metadata = module.MetadataReader;
+        }
+
+        public static bool IsSupported(MetadataReader metadata)
+        {
+            // Only support this map with R2R images of some size
+            return metadata.MethodDefinitions.Count > 32;
+        }
+
+        public override int ClassCode => 606284890;
+
+        protected override string ModuleSpecificName => "__MethodIsGenericMap__";
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            // This node does not trigger generation of other nodes.
+            if (relocsOnly)
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolDefinitionNode[] { this });
+
+            ObjectDataBuilder builder = new ObjectDataBuilder(factory, relocsOnly);
+            builder.AddSymbol(this);
+
+            builder.EmitInt(_metadata.MethodDefinitions.Count);
```

Comment:
Assert 32 limit.

Reply:
Structurally this can't be a problem. Count is limited by the ECMA format to 24bits.